pinecone/test_suite.go

```diff
@@ -17,32 +16,32 @@ import (
 
 type IntegrationTests struct {
 	suite.Suite
-	apiKey           string
-	client           *Client
-	host             string
-	dimension        int32
-	indexType        string
-	vectorIds        []string
-	idxName          string
-	idxConn          *IndexConnection
-	sourceTag        string
-	clientSourceTag  Client
-	idxConnSourceTag *IndexConnection
+	apiKey         string
+	client         *Client
+	host           string
+	dimension      int32
+	indexType      string
+	vectorIds      []string
+	idxName        string
+	idxConn        *IndexConnection
+	collectionName string
+	sourceTag      string
 }
 
 func (ts *IntegrationTests) SetupSuite() {
 	ctx := context.Background()
 
-	additionalMetadata := map[string]string{"api-key": ts.apiKey}
+	_, err := WaitUntilIndexReady(ts, ctx)
+	require.NoError(ts.T(), err)
 
 	namespace, err := uuid.NewUUID()
 	require.NoError(ts.T(), err)
 
-	idxConn, err := newIndexConnection(newIndexParameters{
-		additionalMetadata: additionalMetadata,
-		host:               ts.host,
-		namespace:          namespace.String(),
-		sourceTag:          ""})
+	idxConn, err := ts.client.Index(NewIndexConnParams{
+		Host:      ts.host,
+		Namespace: namespace.String(),
```

Comment:
weird that namespace isn't a string when we make it but whatevs

Reply:
I can tweak that, was using `uuid` slightly differently.